### ISSUE TITLE
Variable parameters

### ISSUE THREAD

**Issue by regul4rj0hn** [aspnet/Docs#7451]
How would you do this using the new &lt;partial /&gt; tags:
```
@foreach (var scope in Model.ResourceScopes)
{
  @Html.Partial("_ScopeListItem", scope)
}
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fa4c23ba-19f6-d847-fefa-a61e283d5fde
* Version Independent ID: e7df465a-5f3b-1522-b2d9-5e3afbf74257
* Content: [Partial Tag Helper in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/built-in/partial-tag-helper?view=aspnetcore-2.1)
* Content Source: [aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/mvc/views/tag-helpers/built-in/partial-tag-helper.md)
* Product: **aspnet-core**
* GitHub Login: @scottaddie
* Microsoft Alias: **scaddie**

**Comment by regul4rj0hn** [comment]
Nevermind. My apologies, missed the @ on the scopes variable. Duh.

**Comment by scottaddie** [comment]
@regul4rj0hn I'll add a "migration from HTML Helpers" section to the doc, as I'm sure others are curious too. In your case, you'll want to use the following:

```csharp
@foreach (var scope in Model.ResourceScopes)
{
    <partial name="_ScopeListItem" model="scope" />
}
```

**Comment by regul4rj0hn** [comment]
Yep, that might help people. It would actually look like this to get it to compile:
```
@foreach (var scope in Model.ResourceScopes)
{
    <partial name="_ScopeListItem" model="@scope" />
}
```
Otherwise, it looks for "scope" inside the current @model tagged for the view, instead of passing in the foreach variable.
Good to know about the async load, thanks!

**Comment by scottaddie** [comment]
@regul4rj0hn It compiles and works for me without the @ symbol before `scope`. If you run `dotnet --version`, what's the version number returned? Also, which version of VS are you using?

**Comment by regul4rj0hn** [comment]
```
> dotnet --version
> 2.1.301
```
VS Professional v17.7.4
https://imgur.com/ufby9ow

**Comment by scottaddie** [comment]
@regul4rj0hn I see the issue. You're assigning `scope` to the `for` attribute instead of the `model` attribute.

**Comment by regul4rj0hn** [comment]
Whoops! You are absolutely correct, that happens when you take a shortcut, and do a find and replace on the @Html.Partial tag across your project to "migrate to 2.1 faster". Thanks for spotting the oversight.
Thanks a lot for your help, Scott!
